### PR TITLE
ref(ddm): Use hasCustomMetrics flag

### PR DESCRIPTION
**Pull request by ArthurKnaus**
- fixes JAVASCRIPT-2QGS

### CODE REVIEW

File: static/app/views/ddm/context.tsx [hunk 1]
```diff
@@ -73,6 +73,17 @@ const emptyWidget: MetricWidgetQueryParams = {
...
+function useCurrentProjects(): Project[] {
+  const {selection} = usePageFilters();
+  const {projects: projectIds} = selection;
+  const {projects} = useProjects();
+
+  return useMemo(() => {
+    const projectIdLookup = new Set(projectIds.map(id => id.toString()));
+    return projects.filter(project => projectIdLookup.has(project.id));
+  }, [projects, projectIds]);
+}
```
Comment (ArthurKnaus) [inline static/app/views/ddm/context.tsx:85]: I have not found an existing util for this, but it feels like there should be one 🤔 
Did I miss something @matejminar ?

Comment (matejminar) [reply static/app/views/ddm/context.tsx:85]: `useProjects` accepts an array of slugs to only fetch/return those - but in this case, we'd need to pass an array of IDs. I am not aware of a better way, unfortunately, our project slug vs project id string vs project id number situation leaves a lot to be desired 😔

**Review by matejminar — APPROVED** [review]
https://github.com/getsentry/sentry/blob/da4c2cc79cf35287d57e6f7a83c69f252e92226f/static/app/views/ddm/layout.tsx#L37
How about we add `hasCustomMetrics` do the condition here? Should result in faster resolution for people that have custom metrics.

**Comment by ArthurKnaus** [comment]
> https://github.com/getsentry/sentry/blob/da4c2cc79cf35287d57e6f7a83c69f252e92226f/static/app/views/ddm/layout.tsx#L37
> 
> 
> How about we add `hasCustomMetrics` do the condition here? Should result in faster resolution for people that have custom metrics.

Makes sense, though we then again have the problem, that the MRI select needs a loading state as it will not be interactive until metrics meta is loaded 🤔

**Review by matejminar — APPROVED** [review]

**Pull request merged**